### ISSUE TITLE
bam empty (problem with temporary bam file?)

### ISSUE THREAD

**Issue by gthilliez** [tseemann/snippy#257]
Hi Thorsten,

I am having some issues with snippy-4.2.1

I am running it on about 100 sequences, and a good proportion of them generate empty bam files.
I used snippy 3.0 on some f those samples in the past and I had no issues with them.

I think the problem might comes from the temporary files 

In the error file attached I get the following lines:
[slurm.err_snippy.txt](https://github.com/tseemann/snippy/files/2577485/slurm.err_snippy.txt)

[bam_sort_core] merging from 1 files and 1 in-memory blocks...
[E::hts_open_format] Failed to open file /tmp/snippy.16204..0000.bam
samtools sort: failed to create temporary file "/tmp/snippy.16204..0000.bam": File exists

I used the snippy --tmpdir to have a look at the temporary files, and they are being generated and they are not empty. But for some reason snippy cannot open them or merge them. Any idea of why this is happening ? Have you seen this issue before?

Thanks
Gaëtan

**Comment by pvishwa2** [comment]
I've encountered this exact problem with while running snippy as well. Have you found out what's causing this issue, and if so what was your solution?

**Comment by PlantDr430** [comment]
I am having the same issue. Has anyone discovered what the root cause was? This only happens on 1 of my samples, but works fine on others.

**Comment by PlantDr430** [comment]
So, I think this is caused by a memory issue. I was able to get it working by raising the --ram option and lowering the --cpus option so that it took advantage of the ram available on our server. This really is in reference to line 341 of the snippy code `my $sort_ram = "-m ".sprintf("%dM", 1000*$ram/$sort_cpus);`. By altering the -m of samtools sort I was able to get the sample working. 

Another way I was able to get my sample to run was --subsample 0.5 as I noticed this sample had about 2x more reads than my other samples that worked. So just lowering the number of reads solved the problem.
